deepinterpolation/cli/training.py

```diff
@@ -0,0 +1,97 @@
+import argschema
+import json
+from pathlib import Path
+
+from deepinterpolation.cli.schemas import TrainingInputSchema
+from deepinterpolation.generic import ClassLoader
+
+
+class Training(argschema.ArgSchemaParser):
+    default_schema = TrainingInputSchema
+
+    def run(self):
+        self.logger.name = type(self).__name__
+
+        outdir = Path(self.args['training_params']['output_path'])
+        if self.args["output_full_args"]:
+            full_args_path = outdir / "training_full_args.json"
+            with open(full_args_path, "w") as f:
+                json.dump(self.args, f, indent=2)
+            self.logger.info(f"wrote {full_args_path}")
+        uid = self.args['run_uid']
+
+        # We create the output model filename if empty
+        if self.args['training_params']["model_string"] == "":
+            self.args['training_params']["model_string"] = (
+                self.args["network_params"]["name"] + "_" +
+                self.args["training_params"]["loss"]
+            )
+
+        # We pass on the uid
+        self.args["training_params"]["run_uid"] = uid
+
+        # We pass on folder
+        self.args["training_params"]["output_dir"] = \
+            self.args['training_params']['output_path']
+
+        # We convert to old schema
+        self.args['training_params']['nb_gpus'] = 2 * \
+            int(self.args['training_params']['multi_gpus'])
+
+        # Forward parameters to the training agent
+        self.args['training_params']['steps_per_epoch'] = \
```

Comment:
Changing `self.args` in the code makes me uneasy. Ideally, `self.args` would just be a record of the arguments passed in by the user. That will especially be helpful if you end up exposing `training_params.steps_per_epoch` to the user in the future. You don't want to silently overwrite your work.

Can you expose `training_params.steps_per_epoch` now, default it to `None`, and, at least, enclose this code in an `if self.args['training_params']['steps_per_epoch'] is None` block (the same goes for all of these lines overwriting or modifying `self.args`?

Reply:
So this parameters is exposed actually. The reason it is passed here is that the intermediate generator and training object needs to have access to this parameter in the current implementation. I am hoping to remove that dependency with a separate PR (as discussed above). Exposing this parameters more than once will force the use to feed the parameter multiple times which cause some errors. I therefore decided to copy it around there.